### PR TITLE
Use a separate Knit tool

### PR DESCRIPTION
**Pull request by elizarov**
Knit is now a separate tool with lots of flexibility, providing cleaner and more configurable use with more features (auto numbering, template-based tests, etc)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -41,7 +41,15 @@ buildscript {
         }
     }
 
-    if (build_snapshot_train || atomicfu_version.endsWith("-SNAPSHOT")) {
+    ext.using_snapshot_version=build_snapshot_train
```
Comment (qwwdfsad) [inline build.gradle]: missing indents

File: knit.properties [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+#
+
+knit.pattern=example-[a-zA-Z0-9-]+-##\\.kt
```
Comment (qwwdfsad) [inline knit.properties]: We have `knit` properties in a root folder, in `docs` and in `knit` project.
Could we please reduce the count of configuration files for Knit?

File: knit/README.md [hunk 3]
```diff
@@ -1,9 +1,6 @@
 # Knit
 
-This is a very simple tool that produces Kotlin source example files from a markdown document that includes
-snippets of Kotlin code in its body. It is used to produce examples for 
-[coroutines guide](../docs/coroutines-guide.md) and other markdown documents.
-It also includes links to the documentation web site into the documents.
+TODO
```
Comment (qwwdfsad) [inline knit/README.md]: TODO should probably be replaced with documentation

File: knit/build.gradle [hunk 4]
```diff
@@ -1,26 +1,13 @@
 /*
- * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
```
Comment (qwwdfsad) [inline knit/build.gradle]: 2020 :)

**Comment by qwwdfsad** [comment]
https://teamcity.jetbrains.com/buildConfiguration/KotlinTools_KotlinxCoroutines_BuildWindows/2764082?buildTab=log&focusLine=3171&linesState=41 build on Windows consistently fails, please investigate

**Comment by elizarov** [comment]
> https://teamcity.jetbrains.com/buildConfiguration/KotlinTools_KotlinxCoroutines_BuildWindows/2764082?buildTab=log&focusLine=3171&linesState=41 build on Windows consistently fails, please investigate

The reason is:

```
  Execution failed for task ':kotlinx-coroutines-android:dokka'. com.google.inject.ProvisionException: Guice provision errors:

1) Error injecting constructor, java.lang.RuntimeException: Exception while loading package-list from ExternalDocumentationLinkImpl(url=https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/, packageListUrl=file://C:/BuildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/build/dokka/kotlinx-coroutines-core/package-list)
  at org.jetbrains.dokka.ExternalDocumentationLinkResolver.<init>(ExternalDocumentationLinkResolver.kt:27)
  at org.jetbrains.dokka.ExternalDocumentationLinkResolver.class(ExternalDocumentationLinkResolver.kt:27)
  while locating org.jetbrains.dokka.ExternalDocumentationLinkResolver
    for parameter 4 at org.jetbrains.dokka.DeclarationLinkResolver.<init>(DeclarationLinkResolver.kt:13)
  while locating org.jetbrains.dokka.DeclarationLinkResolver
    for parameter 0 at org.jetbrains.dokka.PackageDocs.<init>(PackageDocs.kt:13)
  at org.jetbrains.dokka.PackageDocs.class(PackageDocs.kt:12)
  while locating org.jetbrains.dokka.PackageDocs
```

Build script now runs `dokka`, too, as a prerequisite to `knitCheck` (to make sure all committed docs files are file), so it exposes all the dokka problem now. This is good, actually.

File: knit.properties (hunk 2 above)
Comment (elizarov) [reply knit.properties]: I dropped a separate knit module and directory. It is way leaner now. All the core knit configs are in `docs` folder now (where they are actually used).

**Pull request merged**